python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -199,6 +199,24 @@ def handle_output(self, context, obj):
 
     def load_input(self, context):
         """Unpickle the file and Load it to a data object."""
+        if context.has_asset_partitions:
+            return [
+                self._load_pickle(self._partition_path(context.asset_key, partition_key))
```

Comment:
On the GH issue you files, you proposed including the partition keys in the result somehow. Any particular reason you moved away from this?

Reply:
No particular reason, I just went with the minimal implementation to begin with. I've added a key/value dict to make it easier for consuming code to look up fields. Users that want a list of values can use `.values()` method of dict.